spec/unit/manifest_spec.rb

```diff
@@ -92,6 +92,44 @@
         }
       EOT
       expect {
+        manifest.validate!
+      }.not_to raise_error
+    end
+  end
+
+  describe "#validate" do
+    it "shows a warning when validates compatible descriptions" do
+      manifest = Manifest.new("name", <<-EOT)
+        {
+          "meta": {
+            "format_version": 2,
+            "os": "invalid"
+          }
+        }
+      EOT
+
+      expect(Machinery::Ui).to receive(:warn).with(
+        "Warning: System Description validation errors:"
+      )
+      expect(Machinery::Ui).to receive(:warn).with(
+        [
+          "The property '#/meta/os' of type String did not match the following " \
+          "type: object in schema b5a414ed-d79e-5362-92f8-a0640edcc3fd#"
+        ]
+      )
+
+      manifest.validate
+    end
+
+    it "doesn't validate incompatible descriptions" do
+      manifest = Manifest.new("name", <<-EOT)
+        {
+          "meta": {
+            "os": "invalid"
+          }
+        }
+      EOT
+      expect {
         manifest.validate
       }.not_to raise_error
```

Comment:
The check is not consistent with the name of the test. If it doesn't valid it should cause an error.